ui/src/app/domains/views/DomainsList/DomainListHeaderForm/DomainListHeaderForm.tsx

```diff
@@ -0,0 +1,87 @@
+import { Col, Input, Row } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+import * as Yup from "yup";
+import type { SchemaOf } from "yup";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import { actions as domainActions } from "app/store/domain";
+import domainSelectors from "app/store/domain/selectors";
+
+type Props = {
+  closeForm: () => void;
+};
+
+export type CreateDomain = {
+  name: string;
+  isAuthoritative: boolean;
+};
+
+const DomainListHeaderForm = ({ closeForm }: Props): JSX.Element => {
+  const dispatch = useDispatch();
+  const errors = useSelector(domainSelectors.errors);
+  const saved = useSelector(domainSelectors.saved);
+  const saving = useSelector(domainSelectors.saving);
+
+  // Pattern that matches a domainname.
+  // XXX 2016-02-24 lamont: This also matches "example.com.",
+  // which is wrong.
+  const domainnamePattern = /^([a-z\d]|[a-z\d][a-z\d-.]*[a-z\d])*$/i;
+
+  const domainNameSchema: SchemaOf<CreateDomain> = Yup.object().shape({
+    name: Yup.string()
+      .required("Domain name cannot be empty")
+      .matches(domainnamePattern, "The domain name is incorrect")
+      .max(253, "Domain name is too long"),
+    isAuthoritative: Yup.bool(),
+  });
+
+  return (
+    <FormikForm<CreateDomain>
+      buttonsBordered={false}
+      errors={errors}
+      initialValues={{
+        name: "",
+        isAuthoritative: true,
+      }}
+      inline={true}
+      onCancel={closeForm}
+      onSubmit={(values: CreateDomain) => {
+        dispatch(
+          domainActions.create({
+            authoritative: values.isAuthoritative,
+            name: values.name,
+          })
+        );
+      }}
+      resetOnSave={true}
+      saving={saving}
+      saved={saved}
+      submitLabel="Save domain"
+      validationSchema={domainNameSchema}
+    >
+      <Row style={{ width: "50%", marginRight: "auto" }}>
```

Comment:
It looks like your going for a `col-6` within the row.

Reply:
doing something like that wasn't enough
```jsx
      <Row>
        <Col size="6">
          <Row>
            <Col size="6">
              <FormikField
                label="Name"
                type="text"
                name="name"
                placeHolder="Domain name"
                required
              />
            </Col>
            <Col size="6">
              <FormikField
                label="Authoritative"
                type="checkbox"
                name="isAuthoritative"
              />
            </Col>
          </Row>
        </Col>
      </Row>
```
Because of the way the form is set currently we can only affect the "inner" part of the form which doesn't include the buttons
![image](https://user-images.githubusercontent.com/11927929/121317798-eabc5880-c90a-11eb-8c11-bca4e65cf1f6.png)

Caleb has made some changes to the form component though so I'll rebase and update it

Reply:
once you rebase and are able to add a className to the formik form you can add the `u-flex--between` class

Reply:
Yeah I think this is one of those things where we just keep the buttons on a separate div and align them right, to keep it consistent with other forms like this.

Reply:
even with the classname option it wasn't working (because of the inline form overwriting it) so we followed Caleb's advice and kept it simple
![image](https://user-images.githubusercontent.com/11927929/121331453-2d842d80-c917-11eb-9783-baee10a4f27e.png)